contracts/hydro/src/contract.rs

```diff
@@ -487,38 +515,44 @@ fn refresh_single_lock(
         &lock_entry,
         env.block.height,
     )?;
-    let validator_result = validate_denom(
+    let validate_denom_result = token_manager.validate_denom(
         &deps.as_ref(),
         current_round_id,
-        constants,
         lock_entry.funds.denom.clone(),
     );
-    if validator_result.is_err() {
-        return Err(ContractError::Std(StdError::generic_err(
-            "Lock denom is for a validator who is currently not in the set, try refreshing when the validator has enoug delegation",
-        )));
-    }
-    let validator = validator_result.unwrap();
+
+    let token_group_id = match validate_denom_result {
+        Ok(token_group_id) => token_group_id,
+        Err(err) => {
+            return Err(new_generic_error(format!(
```

Comment:
Nit: nested errors should not repeat that they are an error
(since it will lead to things like e.g. "Can't refresh lock. Error: Can't validate denom. Error: Can't find channel 468")

Reply:
I would take inspiration from https://github.com/informalsystems/hydro/blob/dusan/locking-different-tokens/contracts/hydro/src/contract.rs#L325 and have it be "validating denom: {}", error